x/crosschain/keeper/msg_server_vote_outbound_tx.go

```diff
@@ -100,6 +100,11 @@ func (k msgServer) VoteOutbound(
 		return &types.MsgVoteOutboundResponse{}, nil
 	}
 
+	// If the CCTX is in a terminal state, we do not need to process it.
+	if cctx.CctxStatus.Status.IsTerminal() {
```

Comment:
Is there any downstream rewards distribution? If yes, this could block a late voter from getting rewards; if no, it is fine to return early.

Reply:
This section of logic should only be executed once for every CCTX , and this part is reponsible to either 
- Finalize cctx 
- Create a revert 


The only reason when the status can `Aborted` at line 104 is if the CCTX has been aborted using `AbortStuckCCTX` without waiting for all the votes to go through . 
In this condition, I don't think we need to payout rewards for the ballot , wdyt?  
Can you think of any other condition for this clause ?

Reply:
it is fine if there is not rewards distribution.